src/NuGet.Services.Metadata.Catalog.Monitoring/Utility/Json/NuGetVersionConverter.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using NuGet.Versioning;
+
+namespace NuGet.Services.Metadata.Catalog.Monitoring
+{
+    public class NuGetVersionConverter : JsonConverter
+    {
```

Comment:
As far as I can tell, it doesn't support null. I think I could probably leverage it and reduce the complexity of this code here.